test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,280 @@
+
+            expectToContainOnly(combos, ["author-mod-1.0.0"]);
+        });
+    });
+
+
+    describe("Single mod with a dependency", () => {
+        it("should return the mod and its dependency's latest version", async () => {
+            const combos = [modWithDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-mod-2.0.0",
+                "author-modWithDependency-1.0.0"
+            ]);
+        });
+    });
+
+
+    describe("Two mods both having a dependency to the same mod", () => {
+        it("should return the two mods at their specific versions and the dependency's latest version", async () => {
+            const combos = [modWithDependency, modWithTheSameDependency];
+
+            await getFullDependencyList(combos, game, []);
+            
+            expectToContainOnly(combos, [
+                "author-mod-2.0.0",
+                "author-modWithDependency-1.0.0",
+                "author-modWithTheSameDependency-1.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with a deep dependency", () => {
+        it("should return the mod and its dependency's last version and its dependency's dependency's latest version", async () => {
+            const combos = [modWithDeepDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modWithDeepDependency-1.0.0",
+                "author-modWithDependency-2.0.0",
+                "author-mod-2.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with a loop dependency", () => {
+        it("should return the mod's specific version and its dependency's last version", async () => {
+            const combos = [modWithLoopDependencyA];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modWithLoopDependencyA-1.0.0",
+                "author-modWithLoopDependencyB-2.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with two dependencies that have the same dependency", () => {
+        it("should return the mod's specific version and both dependecies' last version and the deep dependant's last version", async () => {
+            const combos = [modWithTwoDependeciesThatHaveTheSameDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modWithTwoDependeciesThatHaveTheSameDependency-1.0.0",
+                "author-modWithDependency-2.0.0",
+                "author-modWithTheSameDependency-2.0.0",
+                "author-mod-2.0.0"
+            ]);
+        });
+    });
+
+    
+
+    describe("Single mod with a dependency to a mod that's already installed", () => {
+        it("should return the mod and its dependency's latest version", async () => {
```

Comment:
- This description doesn't seem to match what's actually tested in `expectToContainOnly`
- How does a modpack behave in the same scenario i.e. should it have it's own test case?

Reply:
Oops, now it is described correctly and I added the case for the mod pack (which for some reason is broken, but I'll fix it)

Reply:
Fixed now.